runtime/gc_glue_java/ArrayletObjectModelBase.hpp

```diff
@@ -233,6 +254,15 @@ class GC_ArrayletObjectModelBase
 			 * 	leaf count = (dataSizeInBytes >> leafLogSize) + 1
 			 */
 			numberOfArraylets = ((dataSizeInBytes >> leafLogSize) + (((dataSizeInBytes & leafSizeMask) + leafSizeMask) >> leafLogSize));
+
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+			/* By forcing the layout to be discontiguous in some special cases the last arraylet leaf may be a 
+			 * NULL pointer. In that case we do not consider such leaf for this arraylet.
+			 */
+			if ((_enableDoubleMapping && 1 < numberOfArraylets) && (0 == unadjustedDataSizeInBytes % leafSize)) {
```

Comment:
need two more sets of () brackets (coding standard that (almost) each binary operator must be in own set of brackets).